code/src/components/app-range-slider/AppRangeSlider.tsx

```diff
@@ -0,0 +1,100 @@
+import { ChangeEvent, useMemo, useState } from "react";
+
+import Slider from "@mui/material/Slider";
+
+import AppBox from "@/components/app-box/AppBox";
+import AppInput from "@/components/app-input/AppInput";
+import { AppRangeSliderProps } from "@/components/app-range-slider/AppRangeSlider.types";
+import AppTypography from "@/components/app-typography/AppTypography";
+
+import cn from "@/utils/cn/cn";
+
+import "@/components/app-range-slider/AppRangeSlider.scss";
+
+const AppRangeSlider = ({
+  className,
+  onChange,
+  min = 0,
+  max = 20000,
+  step = 10,
+  value,
+  ...props
+}: AppRangeSliderProps) => {
+  const initialRangeStart = value?.[0] ?? min;
+  const initialRangeEnd = value?.[1] ?? max
+
+  const [rangeStart, setRangeStart] = useState(initialRangeStart);
+  const [rangeEnd, setRangeEnd] = useState(initialRangeEnd);
+
+  const commonRangeProps = useMemo(
+    () => ({ min, max, step }),
+    [min, max, step]
+  );
+
+  const handleRangeStartChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+    const numericValue = parseInt(value);
+    const limitedNewValue = Math.max(min, numericValue);
+
+    // needed for typescript, we don't need to pass first parameter at all
+    const typedEvent = event as unknown as Event;
+
+    setRangeStart(limitedNewValue);
+    onChange?.(typedEvent, [limitedNewValue, rangeEnd]);
+  };
+
+  const handleRangeEndChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+    const numericValue = parseInt(value);
+    const limitedNewValue = Math.min(max, numericValue);
+
+    // needed for typescript, we don't need to pass first parameter at all
+    const typedEvent = event as unknown as Event;
+
+    setRangeEnd(limitedNewValue);
+    onChange?.(typedEvent, [rangeStart, limitedNewValue]);
+  };
+
+  const handleSliderChange = (event: Event, newValue: number | number[]) => {
+    const typedNewValue = newValue as number[];
+
+    setRangeStart(typedNewValue[0]);
+    setRangeEnd(typedNewValue[1]);
+    onChange?.(event, typedNewValue);
+  };
```

Comment:
```
const value = event.target.value;
const numericValue = parseInt(value);
const limitedNewValue = Math.max(min, numericValue);
```
and some `typedEvent` `onChange` please improve it

Reply:
Created helper function:

```tsx
const parseRangePositionValue = (
  event: ChangeEvent<HTMLInputElement>,
  limiterFn: typeof Math.max | typeof Math.min,
  edgeValue: number
) => {
  const value = event.target.value;
  const numericValue = parseInt(value);
  const limitedNewValue = limiterFn(edgeValue, numericValue);

  // needed for typescript, we don't need to pass first parameter at all
  const typedEvent = event as unknown as Event;

  return { typedEvent, limitedNewValue };
};
```